ocaml/typing/printtyp.ml

```diff
@@ -1420,26 +1413,47 @@ and tree_of_qtvs qtvs =
 and tree_of_row_field mode (l, f) =
   match row_field_repr f with
   | Rpresent None | Reither(true, [], _) -> (l, false, [])
-  | Rpresent(Some ty) -> (l, false, [tree_of_typexp mode ty])
+  | Rpresent(Some ty) -> (l, false, [tree_of_typexp mode Alloc.Const.legacy ty])
   | Reither(c, tyl, _) ->
       if c (* contradiction: constant constructor with an argument *)
       then (l, true, tree_of_typlist mode tyl)
       else (l, false, tree_of_typlist mode tyl)
   | Rabsent -> (l, false, [] (* actually, an error *))
 
 and tree_of_typlist mode tyl =
-  List.map (tree_of_typexp mode) tyl
+  List.map (tree_of_typexp mode Alloc.Const.legacy) tyl
 
 and tree_of_labeled_typlist mode tyl =
-  List.map (fun (label, ty) -> label, tree_of_typexp mode ty) tyl
+  List.map (fun (label, ty) -> label, tree_of_typexp mode Alloc.Const.legacy ty) tyl
 
 and tree_of_typ_gf (ty, gf) =
   let gf =
     match gf with
     | Global_flag.Global -> Ogf_global
     | Global_flag.Unrestricted -> Ogf_unrestricted
   in
-  (tree_of_typexp Type ty, gf)
+  (tree_of_typexp Type Alloc.Const.legacy ty, gf)
+
+and tree_of_ret_typ mode acc_mode (m, ty) =
```

Comment:
There are three mode parameters here. Please write a comment (maybe with an example) explaining what they all are. Probably also good to use labels. As it is, I don't understand this function well enough to be able to review well. (Don't resolve, so that I come back and review.)

Reply:
The first `mode` is unrelated to our mode system. I added some comments; together with the comments inside the function, I hope it's now clearer.

Reply:
Maybe forgot to push? I don't see the comments.